client/webserver/site/src/js/doc.ts

```diff
@@ -625,3 +639,51 @@ function timeMod (t: number, dur: number) {
   const n = Math.floor(t / dur)
   return [n, t - n * dur]
 }
+
+function formatSigFigsWithFormatters (sigFigs: number, intFormatter: Intl.NumberFormat, sigFigFormatter: Intl.NumberFormat, n: number, maxDecimals?: number): string {
+  if (n >= Math.round(Math.pow(10, sigFigs - 1))) return intFormatter.format(n)
+  const s = sigFigFormatter.format(n)
+  if (typeof maxDecimals !== 'number') return s
+  const fractional = sigFigFormatter.formatToParts(n).filter((part: Intl.NumberFormatPart) => part.type === 'fraction')[0].value
+  if (fractional.length <= maxDecimals) return s
+  return fullPrecisionFormatter(maxDecimals).format(n)
+}
+
+if (process.env.NODE_ENV === 'development') {
+  // Code will only appear in dev build.
+  // https://webpack.js.org/guides/production/
+  window.testFormatFourSigFigs = () => {
+    const tests: [string, string, number | undefined, string][] = [
+      ['en-US', '1.234567', undefined, '1.235'], // sigFigFormatter
+      ['en-US', '1.234567', 2, '1.23'], // decimalFormatter
+      ['en-US', '1234', undefined, '1,234.0'], // oneFractionalDigit
+      ['en-US', '12', undefined, '12.00'], // sigFigFormatter
+      ['fr-FR', '123.45678', undefined, '123,5'], // oneFractionalDigit
+      ['fr-FR', '1234.5', undefined, '1 234,5'], // U+202F for thousands separator
+      // For Arabic, https://www.saitak.com/number is useful, but seems to use
+      // slightly different unicode points and no thousands separator. I think
+      // the Arabic decimal separator is supposed to be more like a point, not
+      // a comma, but Google Chrome uses U+066B (Arabic Decimal Separator),
+      // which looks like a comma to me. ¯\_(ツ)_/¯
+      ['ar-EG', '123.45678', undefined, '١٢٣٫٥'],
+      ['ar-EG', '1234', undefined, '١٬٢٣٤٫٠'],
+      ['ar-EG', '0.12345', 3, '٠٫١٢٣']
```

Comment:
So it is test-specific issue (and using `navigator.languages` is what we want users to use in practice), right ? We can use something like this to accommodate this test (it passes now): https://github.com/decred/dcrdex/pull/2278/commits/2577606fed7cf1bc034d1165da05462ed5ebf977

The reason I wanted to remove `preciseFormatter` from there is because it kinda duplicates `maxDecimals` param (according to my understanding) and it isn't clear what happens if `preciseFormatter` has precision set to anything different from `maxDecimals` value.

<img width="600" alt="image" src="https://user-images.githubusercontent.com/112318969/231418263-dd96d7d2-6dfe-4651-a2c7-17b6bee85616.png">